scripts/secnetperf-helpers.psm1

```diff
@@ -3,14 +3,78 @@
     Various helper functions for running secnetperf tests.
 #>
 
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Path to the WER registry key used for collecting dumps on Windows.
+$WerDumpRegPath = "HKLM:\Software\Microsoft\Windows\Windows Error Reporting\LocalDumps\secnetperf.exe"
+
 # Write a GitHub error message to the console.
 function Write-GHError($msg) {
     Write-Host "::error::$msg"
 }
 
-# Write a GitHub warning message to the console.
-function Write-GHWarning($msg) {
-    Write-Host "::warning::$msg"
+# Configured the remote machine to collect dumps on crash.
+function Configure-DumpCollection {
+    param ($Session)
+    if ($isWindows) {
+        Invoke-Command -Session $Session -ScriptBlock {
+            $DumpDir = "C:/_work/quic/artifacts/crashdumps"
+            New-Item -Path $DumpDir -ItemType Directory -ErrorAction Ignore | Out-Null
+            New-Item -Path $Using:WerDumpRegPath -Force -ErrorAction Ignore | Out-Null
+            Set-ItemProperty -Path $Using:WerDumpRegPath -Name DumpFolder -Value $DumpDir | Out-Null
+            Set-ItemProperty -Path $Using:WerDumpRegPath -Name DumpType -Value 2 | Out-Null
+        }
+        $DumpDir = Join-Path (Split-Path $PSScriptRoot -Parent) "artifacts/crashdumps"
+        New-Item -Path $DumpDir -ItemType Directory -ErrorAction Ignore | Out-Null
+        New-Item -Path $WerDumpRegPath -Force -ErrorAction Ignore | Out-Null
+        Set-ItemProperty -Path $WerDumpRegPath -Name DumpFolder -Value $DumpDir | Out-Null
+        Set-ItemProperty -Path $WerDumpRegPath -Name DumpType -Value 2 | Out-Null
+    } else {
+        # TODO: Configure Linux to collect dumps.
+    }
+}
+
+# Collects any crash dumps that were generated locally by secnetperf.
+function Collect-LocalDumps {
+    param ($OutputDir)
+    if ($isWindows) {
+        $DumpFiles = (Get-ChildItem "./artifacts/crashdumps") | Where-Object { $_.Extension -eq ".dmp" }
+        if ($DumpFiles) {
+            mkdir $OutputDir | Out-Null
+            foreach ($File in $DumpFiles) {
+                Copy-Item -Path $File.FullName -Destination $OutputDir
+            }
+            # Delete all the files in the crashdumps folder.
+            Remove-Item -Path "./artifacts/crashdumps/*" -Force
+            return $true
+        }
+    } else {
+    }
+    return $false
+}
+
+# Collect any crash dumps that were generated on the remote machine.
+function Collect-RemoteDumps {
+    param ($Session, $OutputDir)
+    if ($isWindows) {
+        $DumpFiles = Invoke-Command -Session $Session -ScriptBlock {
+            Get-ChildItem "C:/_work/quic/artifacts/crashdumps" | Where-Object { $_.Extension -eq ".dmp" }
```

Comment:
Use globally defined const name?

Reply:
Yes, for our perf machines, this is a fixed path.